### ISSUE TITLE
Create an Self-Contained Example that relies on the installation of Mx

### ISSUE THREAD

**Issue by webern** [webern/mx#78]
Somewhere, probably at the top-level, an Examples directory could be created.
In there a sub directory could contain a small main function along with a CMakeLists.txt file. In that CMakeLists.txt file we should relay on this installation of mx and use `find_package( mx )`. During continuous integration runs, we should add a step where we `make install` then cd to the example and build it. If mx installation isn't working, that build should fail. If mx installation is working, that build should succeed.

**Comment by p-groarke** [comment]
Are you ok with `ExternalProject` instead of `find_package` ? It tests the same thing, but it makes it more robust on Windows, as windows doesn't have a standardized system library directory.

**Comment by webern** [comment]
Yes, basically whatever you are using should be tested so we don’t break it!


On April 4, 2020 at 8:01:56 AM, p-groarke (notifications@github.com) wrote:

Are you ok with ExternalProject instead of find_package ? It tests the same
thing, but it makes it more robust on Windows, as windows doesn't have a
standardized system library directory.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/webern/mx/issues/78#issuecomment-609041540>, or
unsubscribe
